spring-cloud-task-docs/src/main/asciidoc/batch-starter.adoc

```diff
@@ -0,0 +1,532 @@
+
+| `spring.batch.job.jdbccursoritemreader.useSharedExtendedConnection`
+| `boolean`
+|
+| Indicates if the connection is shared with other processing (and therefor part of a transaction)
+
+| `spring.batch.job.jdbccursoritemreader.sql`
+| `String`
+| `null`
+| SQL query to read from.
+|===
+
+For more information about the `JdbcCursorItemReader`, refer to its documentation https://docs.spring.io/spring-batch/docs/4.3.x/api/org/springframework/batch/item/database/JdbcCursorItemReader.html[here]
+
+[[kafkaItemReader]]
+=== KafkaItemReader
+
+Ingesting a partition of data from a Kafka topic is useful and exactly what the
+`KafkaItemReader` can do. In order to configure a `KafkaItemReader`, two pieces
+of configuration are required. First, configuring Kafka via Spring Boot's Kafa
+autoconfiguration is required (you can read more about that
+https://docs.spring.io/spring-boot/docs/2.4.x/reference/htmlsingle/#boot-features-kafka[here]).
+Once the Kafka properties from Spring Boot are configured, the `KafkaItemReader`
+itself can be configured via the following properties.
+
+.`KafkaItemReader` Properties
+|===
+| Property | Type | Default Value | Description
+
+| `spring.batch.job.kafkaitemreader.name`
+| `String`
+| `null`
+| Name used to provide unique keys in the `ExecutionContext`
+
+| `spring.batch.job.kafkaitemreader.topic`
+| `String`
+| `null`
+| Name of the topic to read from.
+
+| `spring.batch.job.kafkaitemreader.partitions`
+| `List<Integer>`
+| empty list
+| List of partition indicies to read from.
+
+| `spring.batch.job.kafkaitemreader.pollTimeOutInSeconds`
+| `long`
+| 30
+| Timeout for the `poll()` operations.
+
+| `spring.batch.job.kafkaitemreader.saveState`
+| `boolean`
+| `true`
+| Determines if the state should be saved for restarts
+|===
+
+You can read more about the `KafkaItemReader` via its documentation
+https://docs.spring.io/spring-batch/docs/4.3.x/api/org/springframework/batch/item/kafka/KafkaItemReader.html[here].
+
+[[item-processors]]
+== ItemProcessor Configuration
+
+The single step batch job autoconfiguration will accept an `ItemProcessor` if one
+is available within the `ApplicationContext`. If one is found of the correct type
+(`ItemProcessor<Map<String, Object>, Map<String, Object>>`, it will be autowired
+into the step.
+
+[[item-writers]]
+== Autoconfiguration for ItemWriter implementations
+
+This starter provides autoconfiguration for `ItemWriter` implementations that
+match those `ItemReader` implementations supported: `AmqpItemWriter`,
+`FlatFileItemWriter`, `JdbcItemWriter`, and `KafkaItemWriter`. This section will
+cover how to use the autoconfiguration to configure a supported `ItemWriter`.
+
+[[amqpitemwriter]]
+=== AmqpItemWriter
+
+To write to a RabbitMQ queue, two sets of configuration are required. First, an
+`AmqpTemplate` is required. The easiest way to get this is via Spring Boot's
+RabbitMQ autoconfiguration. You can read about Spring Boot's RabbitMQ support [here].
```

Comment:
here is not populated:  https://docs.spring.io/spring-boot/docs/2.4.x/reference/htmlsingle/#boot-features-amqp[here]